Jenkinsfile

```diff
@@ -142,25 +37,32 @@ node {
             usernamePassword(credentialsId: 'postgresPaymentsPR', usernameVariable: 'postgresUsername', passwordVariable: 'postgresPassword'),
           ]) {
           def extraCommands = "--values ./helm/ffc-demo-payment-service/jenkins-aws.yaml --set container.messageQueueHost=\"$messageQueueHost\",container.scheduleQueueUser=\"$scheduleQueueUsername\",container.scheduleQueuePassword=\"$scheduleQueuePassword\",container.paymentQueueUser=\"$paymentQueueUsername\",container.paymentQueuePassword=\"$paymentQueuePassword\",postgresExternalName=\"$postgresExternalName\",postgresUsername=\"$postgresUsername\",postgresPassword=\"$postgresPassword\""
-          deployPR(kubeCredsId, registry, imageName, containerTag, extraCommands)
+          defraUtils.deployChart(kubeCredsId, registry, imageName, containerTag, extraCommands)
           echo "Build available for review"
         }
       }
     }
     if (pr == '') {
       stage('Publish chart') {
-        publishChart(imageName)
+        defraUtils.publishChart(registry, imageName, containerTag)
+      }
+      stage('Trigger Deployment') {
+        withCredentials([
+          string(credentialsId: 'JenkinsDeployUrl', variable: 'jenkinsDeployUrl'),
+          string(credentialsId: 'ffc-demo-payment-service-deploy-token', variable: 'jenkinsToken')
+        ]) {
+          defraUtils.triggerDeploy(jenkinsDeployUrl, 'ffc-demo-payment-service-deploy', jenkinsToken, ['chartVersion':'1.0.0'])
```

Comment:
For consistency it would be good to move variables within the pipeline to the top of the file, i.e. `ffc-demo-payment-service-deploy` and the parameters

Reply:
I've moved `ffc-demo-payment-service-deploy` to a variable (`jenkinsDeployJob`) as that definitely makes sense.

I'm not sure what you mean about the parameters. I'm assuming they should eventually become dynamic (otherwise, why set them from the pipeline?) in which case they can't be set at the top. Or am I misunderstanding what you meant?

Reply:
I meant the `['chartVersion':'1.0.0']` parameters passed to the deployment job.

Reply:
That's what I thought. Should that not be set dynamically by the pipeline rather than set at the top?

In any case, the current implementation is in line with the other repos so I think it makes sense to leave it until we review the pipelines under [PSD-99](https://eaflood.atlassian.net/browse/PSD-99).